components/channel_info_modal/channel_info_modal.tsx

```diff
@@ -52,17 +68,17 @@ export default class ChannelInfoModal extends React.PureComponent {
         currentRelativeTeamUrl: PropTypes.string,
     };
 
-    constructor(props) {
+    constructor(props: Props) {
         super(props);
 
         this.state = {show: true};
 
-        this.getHeaderMarkdownOptions = memoizeResult((channelNamesMap) => (
+        this.getHeaderMarkdownOptions = memoizeResult((channelNamesMap: Function) => (
```

Comment:
Instead of declaring this above as private and then defining it in the constructor, it would be better to define it outside of the constructor like  `handleFormattedTextClick`. I'm not entirely sure why it was defined in here to begin with.

Reply:
Ah yea that is a good idea, I did that as well as retyping channelNamesMap to use the actual `channelNamesMap` type